Make examples/storage/littlefs support for ESP-IDF versions below 5.2 (IDFGH-11806)

Augtons opened the pull request:
In ESP-IDF 5.2, **littlefs** became a valid [sub-type](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html#subtype
) for the partition table, with the value of `0x83`, and this example uses it. However, before that, this sub-type was not recognized by default. Therefore, I updated the `CMakeLists.txt` file to set it as a custom sub-type for ESP-IDF versions lower than 5.2, in order to make this example compatible with older ESP-IDF versions.

adokitkat commented:
Hi, thank you for contribution. Changes look good to me except for changes in `tools/ci/check_public_headers_exceptions.txt` file - are those necessary to pass tests?

Augtons commented:
Yes, the pre-commit tool automatically added this annoying change for me. I tried to delete it that day, but it would still be automatically added after deletion. Next, I will try to pull origin/master and rebase it to the latest master, hope everything goes well. Thank you!

Augtons commented:
Hi, @adokitkat. I updated my commit, removed the changes of annoying tools/ci/check_public_headers_exceptions.txt and used git rebase to make my commit history look clear. It has passed the test now.

adokitkat commented:
sha=d68b3c3b2f2e97d0b9ea7a182914e13d3d47605a

adokitkat commented:
sha=d68b3c3b2f2e97d0b9ea7a182914e13d3d47605a

adokitkat commented:
Hi @Augtons. Once again thank you for the contribution. After some consideration we decided it would be better to just backport LittleFS example with appropriate commits instead of merging this PR, which is kind of work-around. I am sorry about this. I have created the backports for release/v5.0 and release/v5.1 branches (however there is one blocker issue for v5.0 branch right now so it needs to wait for another MR right now).

adokitkat commented:
littlefs example is now present in IDF v5.0 and newer.
